Laura/update contributing

LauraLangdon opened the pull request:
To address some of the feedback from and close #82:

> IMHO, it would have been helpful if the CONTRIBUTING.md file:
	1	didn't contain generic information on how github works, and instead just described the particularities of contributing to this repo. As it is, the important stuff gets lost in the noise.
	2	would mention that it's my responsibility, as a contributor, to assign either Laura or Jarrod to review my PR.
	3	said that a formatter is REQUIRED, rather than "strongly recommended". I ignored this part, because I figured, "it's markdown, it doesn't need a formatter", and then was subsequently told that a formatter is required.
	4	would state that UC OSPO people are welcome to just clone and branch without forking. This is my preferred method for collaborating on close-knit teams, but I forked because that's what the document said to do. Then I found out from Laura that the direct clone and branch is okay.

1. I left in the generic info, since it's not necessarily the case that all future UC OSPO Network folks will have used git/GitHub previously.
2. Since I've learned not everyone has the option to assign a reviewer, I didn't specify this.
3. I removed this part since it's Python-specific, and added instructions for running Prettier.
4. Similarly to (2), I left this as-is since it turns out not everyone has the permissions to direct clone.

Thanks very much for this feedback, @virginiascarlett! 🙌

virginiascarlett commented:
Thanks Laura, this looks better to me. I have two comments:

- I hadn't looked at the basic git stuff closely, but upon inspection... is their some reason why it directs the reader to name their fork's remote with their username, and to name the upstream remote "origin"? I thought [standard practice](https://docs.github.com/en/get-started/git-basics/managing-remote-repositories) was to call the remote for the main public repo "upstream", while the person's individual fork is called "origin". Some examples [here](https://medium.com/@nynptel/difference-between-origin-and-upstream-for-a-git-repository-67e0a7f1a90e) and [here](https://blog.scottlowe.org/2015/01/27/using-fork-branch-git-workflow/). That last link from Scott Lowe is the way I always do it.
- Maybe if you don't want to get rid of the stuff for beginners, you could add a quickstart for experienced contributors? For example, you could add this to the top of the document:

## Quickstart for experienced GitHub users
    - To preview your changes to the site:
      - [Install Hugo](https://gohugo.io/installation/)
      - [Install Dart Sass](https://gohugo.io/functions/css/sass/#dart-sass)
      - Run `make serve`
    - To make sure your code is formatted as the linter expects, please either:
      - [Run Prettier from your terminal](https://prettier.io/docs/install) or as an extension in your IDE, e.g. the [VS Code Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
      - Wait until you've pushed your code and if the `pre-commit.ci` test fails, add a comment on your PR with the text `pre-commit.ci autofix`. This will tell the pre-commit bot to push a fix, and should work for common formatting issues.
    - Please link to relevant issues in your PR by using [keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue) in the PR description or commit message.
    - We squash and merge PRs, so rebasing is not necessary.

virginiascarlett commented:
Oh, and I've been finding that "😎 Deploy preview" link from netlify extremely useful. I think it's also worth pointing that out in the "Quickstart for experienced GitHub users".

virginiascarlett commented:
Oh, and I'm not sure if this is worth putting in the quickstart because it's such a tiny thing, but... my VSCode Prettier extension does not trim trailing whitespaces, at least not by default. The pre-commit.ci thingy was failing my PR due to trailing whitespaces even after reformatting with prettier. I had to configure VSCode to trim trailing whitespace on save: Settings --> type Files: Trim Trailing Whitespace into the search bar --> Check the box next to "When enabled, will trim trailing whitespace when saving a file."

LauraLangdon commented:
> Oh, and I've been finding that "😎 Deploy preview" link from netlify extremely useful. I think it's also worth pointing that out in the "Quickstart for experienced GitHub users".

Added!

> Oh, and I'm not sure if this is worth putting in the quickstart because it's such a tiny thing, but... my VSCode Prettier extension does not trim trailing whitespaces, at least not by default. The pre-commit.ci thingy was failing my PR due to trailing whitespaces even after reformatting with prettier. I had to configure VSCode to trim trailing whitespace on save: Settings --> type Files: Trim Trailing Whitespace into the search bar --> Check the box next to "When enabled, will trim trailing whitespace when saving a file."

I've added a note to make sure to read the docs on automatically or manually formatting for whichever option a person chooses.
